src/acktest/bootstrapping/s3.py

```diff
@@ -0,0 +1,48 @@
+import boto3
+import logging
+import json
+import re
+import time
+
+from dataclasses import dataclass, field
+from typing import List
+
+from . import BootstrappableResource
+from .. import resources
+from ..aws.identity import get_region, get_account_id
+
+@dataclass
+class Bucket(BootstrappableResource):
+    # Inputs
+    name_prefix: str
+
+    # Outputs
+    name: str = field(init=False)
+    
+    def bootstrap(self):
+        """Creates an S3 bucket with an auto-generated name.
+        """
+        region = get_region()
```

Comment:
might be useful in the future to have `region` be a field on the Bucket resource instead of always pulled from `get_region`. This way, multi-region resources can be created in the same bootstrapping set.

Reply:
Best I can do at the moment is to have a `@property` field that can be overridden. `dataclass` does some funky stuff when you add default value fields in the subclass and it throws an error. There are work arounds but they're ugly. See https://stackoverflow.com/questions/51575931/class-inheritance-in-python-3-7-dataclasses